src/main/java/io/javalin/PathBuilder.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Javalin - https://javalin.io
+ * Copyright 2017 David Åse
+ * Licensed under Apache 2.0: https://github.com/tipsy/javalin/blob/master/LICENSE
+ */
+
+package io.javalin
+
+import java.net.URLEncoder
+
+class PathBuilder(private val baseUrl: String) {
+
+    private val queryParams = mutableMapOf<String, Array<out String>>()
+    private val pathParams = mutableMapOf<String, String>()
+
+    fun pathParam(key: String, value: String): PathBuilder {
+        if (!baseUrl.contains(key)) {
```

Comment:
PathParser + extract params could verify it better

Reply:
Thanks, I'll try that.